### ISSUE TITLE
Segmentation fault when running gen_data_flow_ops.dynamic_stitch

### ISSUE THREAD

**Issue by nimashiri** [tensorflow/tensorflow#59173]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

Yes

### Source

source

### Tensorflow Version

2.10.0

### Custom Code

Yes

### OS Platform and Distribution

Ubuntu 22.04

### Python version

3.9

### Current Behaviour?

```shell
Segmentation fault probably due to dimension mismatch.
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import numpy as np
from tensorflow.python.ops import gen_data_flow_ops
try:
  try:
    with tf.device('/CPU'):
      arg_0_0_tensor = tf.random.uniform([], minval=-256, maxval=257, dtype=tf.int32)
      arg_0_0 = tf.identity(arg_0_0_tensor)
      arg_0_1_tensor = tf.random.uniform([1], minval=-256, maxval=257, dtype=tf.int32)
      arg_0_1 = tf.identity(arg_0_1_tensor)
      arg_0 = [arg_0_0,arg_0_1,]
      arg_1_0_tensor = tf.random.uniform([], minval=-256, maxval=257, dtype=tf.int32)
      arg_1_0 = tf.identity(arg_1_0_tensor)
      arg_1_1_tensor = tf.random.uniform([1], minval=-256, maxval=257, dtype=tf.int32)
      arg_1_1 = tf.identity(arg_1_1_tensor)
      arg_1 = [arg_1_0,arg_1_1,]
      out = gen_data_flow_ops.dynamic_stitch(arg_0,arg_1,)
  except Exception as e:
    print("Error:"+str(e))
  try:
    with tf.device('/GPU:0'):
      arg_0_0 = tf.identity(arg_0_0_tensor)
      arg_0_0 = tf.cast(arg_0_0, tf.int32)
      arg_0_1 = tf.identity(arg_0_1_tensor)
      arg_0_1 = tf.cast(arg_0_1, tf.int32)
      arg_0 = [arg_0_0,arg_0_1,]
      arg_1_0 = tf.identity(arg_1_0_tensor)
      arg_1_0 = tf.cast(arg_1_0, tf.int32)
      arg_1_1 = tf.identity(arg_1_1_tensor)
      arg_1_1 = tf.cast(arg_1_1, tf.int32)
      arg_1 = [arg_1_0,arg_1_1,]
      gen_data_flow_ops.dynamic_stitch(arg_0,arg_1,)
  except Exception as e:
    print("Error:"+str(e))
except Exception as e:
  print("Error:"+str(e))
```
```


### Relevant log output

```shell
2023-01-08 19:41:09.950654: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-08 19:41:09.955250: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-08 19:41:09.955367: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-08 19:41:09.955670: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-01-08 19:41:09.956097: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-08 19:41:09.956203: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-08 19:41:09.956296: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-08 19:41:10.309982: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-08 19:41:10.310131: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-08 19:41:10.310228: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-08 19:41:10.310311: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1616] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 4220 MB memory:  -> device: 0, name: NVIDIA GeForce GTX 1660 Ti, pci bus id: 0000:01:00.0, compute capability: 7.5
Error:{{function_node __wrapped__DynamicStitch_N_2_device_/job:localhost/replica:0/task:0/device:CPU:0}} indices[0] is out of range [Op:DynamicStitch]
Segmentation fault
```
```
</details>

**Comment by tilakrayal** [comment]
@nimashiri,
I was able to execute the given code without any issue/crash on tensorflow v2.11. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/7e02b71b31b70009e8353231b0ee6fc6/untitled852.ipynb) and let us know if it serves the purpose. Thank you!

**Comment by nimashiri** [comment]
> @nimashiri, I was able to execute the given code without any issue/crash on tensorflow v2.11. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/7e02b71b31b70009e8353231b0ee6fc6/untitled852.ipynb) and let us know if it serves the purpose. Thank you!

On 2.11 it should work fine, the issue is for 2.10.

**Comment by tilakrayal** [comment]
@nimashiri,
As it was the bug on tensorflow v2.10, it has been resolved on the latest stable tensorflow v2.11 where we can observe that Segfault error has not been raised. Thank you!
